pybricks/common/pb_type_speaker.c

```diff
@@ -22,34 +23,90 @@
 #include <pybricks/util_mp/pb_obj_helper.h>
 #include <pybricks/util_pb/pb_error.h>
 
+#define MP_PI MICROPY_FLOAT_CONST(3.14159265358979323846)
+
 typedef struct {
     mp_obj_base_t base;
     bool intialized;
 } pb_type_Speaker_obj_t;
 
 STATIC pb_type_Speaker_obj_t pb_type_Speaker_singleton;
 
+STATIC uint8_t volume = 100; // volume in 0..100 range
+
+// The number to multiply the sample amplitude by, to amplify the signal by the defined volume.
+// The original sample amplitude must be in the -1..1 range.
+STATIC uint16_t sample_amplifier = INT16_MAX;
+
 STATIC uint16_t waveform_data[128];
 
+// The amount of radians each PCM sample represents to completely fill whe waveform buffer for 2*PI
+//STATIC mp_float_t pcm_sample_size_rad = MP_PI * 2 / MP_ARRAY_SIZE(waveform_data);
+
+STATIC mp_obj_t pb_type_Speaker_get_volume(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    return mp_obj_new_int(volume);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_KW(pb_type_Speaker_get_volume_obj, 1, pb_type_Speaker_get_volume);
+
+STATIC mp_obj_t pb_type_Speaker_set_volume(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    PB_PARSE_ARGS_METHOD(n_args, pos_args, kw_args,
+        pb_type_Speaker_obj_t, self,
+        PB_ARG_DEFAULT_INT(volume, 100));
+
+    (void)self; // unused
+    mp_int_t new_volume = pb_obj_get_int(volume_in);
+
+    if (volume <= 100) {
+        volume = new_volume;
+    } else {
+        volume = 100;
+    }
```

Comment:
This can be replaced by the helper function `pb_obj_get_pct()`

Reply:
Done.